newIDE/app/src/MainFrame/Toolbar/PreviewButtons.js

```diff
@@ -67,26 +67,29 @@ export default class PreviewButtons extends React.Component<
               />
             }
             openMenuWithSecondaryClick
-            buildMenuTemplate={() => [
-              {
-                type: 'checkbox',
-                label: previewFirstSceneName
-                  ? 'Use scene ' + previewFirstSceneName + ' for preview'
-                  : 'Use this scene for preview',
-                checked: isPreviewFirstSceneOverriden,
-                click: () => {
-                  togglePreviewFirstSceneOverride();
-                  if (!previewFirstSceneName) {
-                    useSceneAsPreviewFirstScene();
-                  }
+            buildMenuTemplate={() => {
+              if (isPreviewFirstSceneOverriden === null) return [];
```

Comment:
> Where should I look though? I've seen you passed `emptyPreviewButtonSettings`, but is this making the menu empty?

Yep, i don't know really what todo.
Here the array is empty, then the menu is empty too. I  can't return null, for not see the menu, because return null crash the app.
I can't wrap `buildMenuTemplate={() => { ... }}` with a condition, because it's not allowed? (or i don't know the syntax).

Reply:
Here the menu empty, but no ide for avoid this empty square.
![image](https://user-images.githubusercontent.com/1670670/76366237-c057a600-6329-11ea-9ff4-ec772147e313.png)

Reply:
Right I missed this when reading the code! So here, more than avoiding an empty square, you want no menu at all.
The problem is that `ElementWithMenu` is not made for not having a menu. If we remove the "menu" from "ElementWithMenu", we would just have... an "Element" :) This is exactly what we're going to do.

Put the `ToolbarIcon`  in its own variable:

```
const previewIcon = <ToolbarIcon
                onClick={onPreview}
                src={
                  isPreviewFirstSceneOverriden
                    ? 'res/ribbon_default/previewOverride32.png'
                    : 'res/ribbon_default/preview32.png'
                }
                tooltip={
                  isPreviewFirstSceneOverriden
                    ? t`Preview is overridden, right click for more`
                    : t`Launch a preview of the scene, right click for more`
                }
              />;
```

then you can either render it directly, or use ElementWithMenu if you want a menu:
```
    return (
      <React.Fragment>
        {showPreviewButton && (
          shouldThereBeAMenu ?
          <ElementWithMenu element={previewIcon} ... /> :
          previewIcon
          )}
        ...
```

You could arguably extract more stuff into variables to make this readable. I supposed here you have a boolean shouldThereBeAMenu - but the name and content of this depends on what you use to decide for the menu presence :)